lib/routing.mixin.ts

```diff
@@ -0,0 +1,98 @@
+
+import BasicRoutingInterface from './routing-interface.js';
+import type { Context } from './page.js';
+import type RouteData from './route-data.js';
+import type RouteTreeNode from './route-tree-node.js';
+import type { LitElement } from 'lit';
+
+export type Constructor<T = {}> = new (...args: any[]) => T;
+export declare class RoutingMixinInterface {
+  routeEnter(currentNode: RouteTreeNode, nextNodeIfExists: RouteTreeNode | undefined, routeId: string, context: Context): Promise<boolean | void>;
+  routeExit(currentNode: RouteTreeNode, nextNode: RouteTreeNode | undefined, routeId: string, context: Context): Promise<boolean | void>;
+}
+const RoutingMixin = <T extends Constructor<LitElement>>(superclass: T) => {
```

Comment:
Same here. Should this be specific to `LitElement`? 
```suggestion
const RoutingMixin = <T extends Constructor<HTMLElement>>(superclass: T) => {
```

Reply:
This was something we discovered in origin. If we don't use LitElement as the super class we get linter errors when using the lifecycle methods like connectedCallback etc

https://lit.dev/docs/composition/mixins/#mixins-in-typescript